High probability of failure maven-enforcer-plugin:3.2.1:enforce (enforce-banned-dependencies) in multi-thread building

## Bug Report

### Which version of ShardingSphere did you use?
0798ec3194771f6c61f679cb14ec6c525a75bad0

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

```bash
./mvnw clean install -T1C -DskipTests
```

My CPU is 16 threads, so Maven `-T1C` means using 16 threads for building.

The following commands could increase the probability of reproducing.
```bash
./mvnw validate -T1C -DskipTests
```

### Expected behavior

Build succeeded stably.

### Actual behavior

The failure modules are different in different builds.

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.724 s (Wall Clock)
[INFO] Finished at: 2023-04-06T10:43:15+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce (enforce-banned-dependencies) on project shardingsphere-data-pipeline-cdc-protocol: Execution enforce-banned-dependencies of goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce failed: Unable to provision, see the following errors:
[ERROR] 
[ERROR] 1) Error in custom provider, java.lang.IllegalStateException
[ERROR]   at org.apache.maven.session.scope.internal.SessionScopeModule.configure(SessionScopeModule.java:64) (via modules: org.eclipse.sisu.wire.WireModule -> org.apache.maven.session.scope.internal.SessionScopeModule)
[ERROR]   while locating org.apache.maven.execution.MavenSession
[ERROR] 
[ERROR] 1 error
[ERROR] -> [Help 1]
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce (enforce-banned-dependencies) on project shardingsphere-transaction-base: Execution enforce-banned-dependencies of goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce failed: Unable to provision, see the following errors:
[ERROR] 
[ERROR] 1) Error in custom provider, java.lang.IllegalStateException
[ERROR]   at org.apache.maven.session.scope.internal.SessionScopeModule.configure(SessionScopeModule.java:64) (via modules: org.eclipse.sisu.wire.WireModule -> org.apache.maven.session.scope.internal.SessionScopeModule)
[ERROR]   while locating org.apache.maven.execution.MavenSession
[ERROR] 
[ERROR] 1 error
[ERROR] -> [Help 1]
```

```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.759 s (Wall Clock)
[INFO] Finished at: 2023-04-06T10:51:30+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce (enforce-maven-version) on project shardingsphere-sql-federation-executor-core: Execution enforce-maven-version of goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce failed: Unable to provision, see the following errors:
[ERROR] 
[ERROR] 1) Error in custom provider, java.lang.IllegalStateException
[ERROR]   at org.apache.maven.session.scope.internal.SessionScopeModule.configure(SessionScopeModule.java:64) (via modules: org.eclipse.sisu.wire.WireModule -> org.apache.maven.session.scope.internal.SessionScopeModule)
[ERROR]   while locating org.apache.maven.execution.MavenSession
[ERROR] 
[ERROR] 1 error
[ERROR] -> [Help 1]
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce (enforce-maven-version) on project shardingsphere-it-optimizer: Execution enforce-maven-version of goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce failed: Unable to provision, see the following errors:
[ERROR] 
[ERROR] 1) Error in custom provider, java.lang.IllegalStateException
[ERROR]   at org.apache.maven.session.scope.internal.SessionScopeModule.configure(SessionScopeModule.java:64) (via modules: org.eclipse.sisu.wire.WireModule -> org.apache.maven.session.scope.internal.SessionScopeModule)
[ERROR]   while locating org.apache.maven.execution.MavenSession
[ERROR] 
[ERROR] 1 error
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :shardingsphere-sql-federation-executor-core
```

--- a/.mvn/wrapper/maven-wrapper.properties
+++ b/.mvn/wrapper/maven-wrapper.properties
@@ -15,5 +15,5 @@
 # limitations under the License.
 #
 
-distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.5/apache-maven-3.8.5-bin.zip
-wrapperUrl=https://repo.maven.apache.org/maven2/org/apache/maven/wrapper/maven-wrapper/3.1.0/maven-wrapper-3.1.0.jar
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.7/apache-maven-3.8.7-bin.zip
+wrapperUrl=https://repo.maven.apache.org/maven2/org/apache/maven/wrapper/maven-wrapper/3.1.1/maven-wrapper-3.1.1.jar

